crates/worker/src/executor/parameter_server.rs

```diff
@@ -137,49 +137,55 @@ impl JobExecutor for ParameterServerExecutor {
 
                 loop {
                     let next_item = tokio::select! {
-                        _ = cancel.cancelled() => None,
+                        _ = cancel.cancelled() => {
+                            break;
+                        },
                         item = incoming.next() => item,
                     };
-                    let Some(item_result) = next_item else { break; };
+                    let Some(item_result) = next_item else { continue; };
```

Comment:
Let's keep this code as it was before, otherwise we won't stop looping if there aren't any items left in `incoming` due to `next_item else { continue; }`. This could be the case if the receiver stream is closed unexpectedly.